src/nvim/syntax.c

```diff
@@ -6759,7 +6759,16 @@ void set_hl_group(int id, HlAttrs attrs, Dict(highlight) *dict, int link_id)
   };
 
   for (int j = 0; cattrs[j].dest; j++) {
-    if (cattrs[j].val != -1) {
+    if (cattrs[j].val == -1) {
+      // Should we clear?
+      if ((cattrs[j].name.type == kObjectTypeString
```

Comment:
This extra check should be unnecessary, just `*cattrs[j].dest = NULL` directly when `cattrs[j].val == -1`

Reply:
Depends what behaviour we want. I've done this so not passing an attr leaves the original value untouched. Only by passing a negative integer or `'NONE'` clears the attr.

Reply:
that's inconsistent with the behavior for `ns>0` tho. The idea of the API is that you can just set the value of a highlight group without worrying whatever was there before.

Reply:
Ok fair enough. I'm not too familiar with `ns>0` behaviour.